skimage/feature/_hoghistogram.pyx

```diff
@@ -0,0 +1,130 @@
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: wraparound=False
+
+import numpy as np
+cimport numpy as cnp
+
+cdef float cell_hog(cnp.float64_t[:, :] magnitude,
+        cnp.float64_t[:, :] orientation,
+        float orientation_start, float orientation_end,
+        int cell_columns, int cell_rows,
+        int column_index, int row_index,
+        int size_columns, int size_rows):
+    """Calculation of the cell's HOG value
+
+    Parameters
+    ----------
+    magnitude : ndarray
+        The gradient magnitudes of the pixels.
+    orientation : ndarray
+        Lookup table for orientations.
+    orientation_start : float
+        Orientation range start.
+    orientation_end : float
+        Orientation range end.
+    cell_columns : int
+        Pixels per cell (x).
+    cell_rows : int
+        Pixels per cell (y).
+    column_index : int
+        Block column index.
+    row_index : int
+        Block row index.
+    size_columns : int
+        Number of columns.
+    size_rows : int
+        Number of rows.
+
+    Returns
+    -------
+    total : float
+        The total HOG value.
+    """
+    cdef int cell_column, cell_row
+
+    cdef float total = 0.
+    for cell_row in range(-cell_rows/2, cell_rows/2):
+        for cell_column in range(-cell_columns/2, cell_columns/2):
+            if (row_index + cell_row < 0
+                or row_index + cell_row >= size_rows
+                or column_index + cell_column < 0
+                or column_index + cell_column >= size_columns
+                or orientation[row_index + cell_row, column_index + cell_column]
```

Comment:
I'm a big believer in "precalculate and reference" anytime a value gets used more than once. These conditions make use of a these intermediate values several times: 

```
row_index + cell_row  # precalculate in the first for loop, reference thereafter
column_index + cell_column  # precalculate at the beginning of the second for loop, reference thereafter
```

Unless Cython is being smart enough to do this on its own, which I doubt, it will result in efficiency gains.